### DISCUSSION TITLE
How to use a python reserved keyword as API parameter name

### DISCUSSION THREAD

**Discussion by frodrigo** [fastapi/fastapi#5184] [Questions] [answered]
I am porting an existing API to FastAPI. But I stuck on a issue with a query parameter named `class`.

I can use raw request to access the parameter. But there is a more or less clean way to do it while keep it on the swagger spec.

(https://github.com/osm-fr/osmose-frontend/issues/384)

**Comment by frankie567** [comment, marked as answer]
You could use the [ `alias` parameter](https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#alias-parameters) of `Query`:

```py
from fastapi import Query

@app.get("/items")
async def list_items(class_parameter: Optional[str] = Query(None, alias="class")):
    ...
```

**Reply by frodrigo** [reply]
Nice. Thank.
